local/network.go

```diff
@@ -467,23 +464,26 @@ func (ln *localNetwork) Healthy(ctx context.Context) error {
 	}(ctx)
 
 	errGr, ctx := errgroup.WithContext(ctx)
-	for _, lnNode := range ln.nodes {
-		lnNode := lnNode
+	for _, node := range ln.nodes {
+		node := node
+		nodeName := node.GetName()
 		errGr.Go(func() error {
 			// Every [healthCheckFreq], query node for health status.
 			// Do this until ctx timeout or network closed.
 			for {
-				if lnNode.Status() != node.Started {
-					return fmt.Errorf("unexpected stop on node %q", lnNode.GetName())
+				if node.Status() != status.Running {
+					// If we had stopped this node ourselves, it wouldn't be in [ln.nodes].
+					// Since it is, it means the node stopped unexpectedly.
```

Comment:
Because we hold the lock here, either:
1. `ln.Stop` was previously called, in which case we return `ErrStopped` above.
2. `ln.Stop` wasn't previously called, in which case it can't execute until after `Healthy` returns. So no nodes in `ln.nodes` should be stopped.